From 7d090a724834dc794d6d56b0b614b46154c9d1c3 Mon Sep 17 00:00:00 2001
From: Adam Duskett <aduskett@gmail.com>
Date: Thu, 28 Oct 2021 13:10:48 -0700
Subject: [PATCH] 2

Signed-off-by: Adam Duskett <aduskett@gmail.com>
---
 CMakeLists.txt | 33 +++++++++++++++++++++++----------
 1 file changed, 23 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3a2daa5..ffcf4a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,6 @@ find_package(Threads)
 #########################################
 set_property(GLOBAL PROPERTY PROTOBUF_USE_STATIC_LIBS ON)  #this flag doesn't actually seem to work yet but it's documented
 find_package(Protobuf 3.17.3 REQUIRED)
-string(REPLACE ${CMAKE_SHARED_LIBRARY_SUFFIX} ${CMAKE_STATIC_LIBRARY_SUFFIX} Protobuf_LITE_STATIC_LIBRARY ${Protobuf_LITE_LIBRARY})
 include_directories(${Protobuf_INCLUDE_DIRS})
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR}) #needed to include generated protobuf headers
@@ -69,15 +68,29 @@ endif(BUILD_TESTS)
 #########################################
 # Boost dependencies                    #
 #########################################
-set_property(GLOBAL PROPERTY Boost_USE_STATIC_LIBS ON)
 set_property(GLOBAL PROPERTY Boost_USE_DEBUG_RUNTIME OFF)
 #set_property(GLOBAL PROPERTY Boost_USE_MULTITHREADED ON)
 find_package(Boost 1.76.0 REQUIRED COMPONENTS system log log_setup thread program_options date_time filesystem)
 include_directories(${Boost_INCLUDE_DIRS})
-foreach(BOOST_LIB ${Boost_LIBRARIES})
-    string(REPLACE ${CMAKE_SHARED_LIBRARY_SUFFIX} ${CMAKE_STATIC_LIBRARY_SUFFIX} BOOST_STATIC_LIB ${BOOST_LIB})
-    list(APPEND Boost_STATIC_LIBRARIES ${BOOST_STATIC_LIB})
-endforeach()
+
+if (BUILD_SHARED_LIBS)
+  set_property(GLOBAL PROPERTY Boost_USE_STATIC_LIBS OFF)
+  set(Boost_USE_STATIC_RUNTIME OFF)
+  ADD_DEFINITIONS(-DBOOST_LOG_DYN_LINK)
+  foreach(BOOST_LIB ${Boost_LIBRARIES})
+      list(APPEND Boost_LINK_LIBRARIES ${BOOST_LIB})
+  endforeach()
+  list(APPEND Protobuf_LINK_LIBRARY ${Protobuf_LITE_LIBRARY})
+else(BUILD_SHARED_LIBS)
+  set_property(GLOBAL PROPERTY Boost_USE_STATIC_LIBS ON)
+  set(Boost_USE_STATIC_RUNTIME ON)
+  foreach(BOOST_LIB ${Boost_LIBRARIES})
+      string(REPLACE ${CMAKE_SHARED_LIBRARY_SUFFIX} ${CMAKE_STATIC_LIBRARY_SUFFIX} BOOST_STATIC_LIB ${BOOST_LIB})
+      list(APPEND Boost_STATIC_LIBRARIES ${BOOST_STATIC_LIB})
+  endforeach()
+  find_package(ZLIB REQUIRED)
+  string(REPLACE ${CMAKE_SHARED_LIBRARY_SUFFIX} ${CMAKE_STATIC_LIBRARY_SUFFIX} Protobuf_LINK_LIBRARY ${Protobuf_LITE_LIBRARY})
+endif(BUILD_SHARED_LIBS)
 
 #########################################
 # Target : Build aws-tunnel-local-proxy #
@@ -109,8 +122,8 @@ endif()
 target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} ${CMAKE_THREAD_LIBS_INIT})
 target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} OpenSSL::SSL)
 target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} OpenSSL::Crypto)
-target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} ${Boost_STATIC_LIBRARIES})
-target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} ${Protobuf_LITE_STATIC_LIBRARY})
+target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} ${Boost_LIBRARIES})
+target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} ${Protobuf_LINK_LIBRARY})
 
 
 include_directories(${PROJECT_SOURCE_DIR}/src)
@@ -118,8 +131,8 @@ if(BUILD_TESTS)
   target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TEST_NAME} ${CMAKE_THREAD_LIBS_INIT})
   target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TEST_NAME} OpenSSL::SSL)
   target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TEST_NAME} OpenSSL::Crypto)
-  target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TEST_NAME} ${Boost_STATIC_LIBRARIES})
-  target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TEST_NAME} ${Protobuf_LITE_STATIC_LIBRARY})
+  target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TEST_NAME} ${Boost_LIBRARIES})
+  target_link_libraries(${AWS_TUNNEL_LOCAL_PROXY_TEST_NAME} ${Protobuf_LINK_LIBRARY})
 endif(BUILD_TESTS)
 
 set_property(TARGET ${AWS_TUNNEL_LOCAL_PROXY_TARGET_NAME} APPEND_STRING PROPERTY COMPILE_FLAGS ${CUSTOM_COMPILER_FLAGS})
-- 
2.32.0

