From 9d6d88b6f48d9f18779b7de28c0493bf1e88da70 Mon Sep 17 00:00:00 2001
From: Adam Duskett <aduskett@rivian.com>
Date: Mon, 21 Jun 2021 09:09:20 -0700
Subject: [PATCH] Make tests optional

Signed-off-by: Adam Duskett <aduskett@rivian.com>
---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f7c676..abc15cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -199,6 +199,5 @@ if (BUILD_TEST_DEPS)
     # and gmock_main
     add_subdirectory(${CMAKE_BINARY_DIR}/googletest-src
             ${CMAKE_BINARY_DIR}/googletest-build)
+    add_subdirectory(test)
 endif ()
-
-add_subdirectory(test)
-- 
2.31.1

